src/main/java/dev/ebrydeu/spring_boot_library/services/MessageService.java

```diff
@@ -0,0 +1,66 @@
+package dev.ebrydeu.spring_boot_library.services;
+import dev.ebrydeu.spring_boot_library.domain.dto.MessageDto;
+import dev.ebrydeu.spring_boot_library.domain.entities.Message;
+import dev.ebrydeu.spring_boot_library.exception.CustomExceptions;
+import dev.ebrydeu.spring_boot_library.repositories.MessageRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.cache.annotation.Cacheable;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service
+    @Transactional
+    public class MessageService {
+        MessageRepository messageRepository;
+
+        @Autowired
+        public MessageService(MessageRepository messageRepository) {
+            this.messageRepository = messageRepository;
+        }
```

Comment:
One thing about `@Autowired` is that it simply annotates that the Dependency Injection (DI) being used is a Bean. In modern Spring Boot, you don't need to declare `@Autowired` if your DI is considered a Bean. But how do you determine if it's a bean or not? If you have IntelliJ Ultimate, it will show a gutter icon (a small bean icon on the left) for beans. But how can you detect this without the help of IntelliJ? It's quite simple but perhaps tedious. You just need to memorize the most common annotations, such as `@Service`, `@Repository`,`@RestController`, `@Controller` .

So in our case, if you open `@Service`, you would see the `@Component` annotation, which is kind of a sibling. What it does is it automatically finds and does `automatic wiring` (essentially the same as what `@Autowired` does, but you don't need to annotate).

But in terms of when we actually need to use `@Autowired`, it's quite simple,  `in tests`.

Nevertheless, even with this, we use Lombok and we don't need a constructor here. We can use just one annotation, which is `@RequiredArgsConstructor`, which does the same job, but only when we have `final` on the field. Also change for `UserService`

```suggestion
@RequiredArgsConstructor
    public class MessageService {
       final MessageRepository messageRepository;
```

Reply:
Its removed, thanks for easy to understand info :)